Marlin/src/module/tool_change.cpp

```diff
@@ -1051,14 +1162,17 @@ void tool_change(const uint8_t new_tool, bool no_move/*=false*/) {
       TEMPORARY_BED_LEVELING_STATE(false);
     #endif
 
-    // First tool priming. To prime again, reboot the machine.
+    // First tool priming. To prime again, reboot the machine. -- Should only occur for first T0 after powerup!
     #if ENABLED(TOOLCHANGE_FS_PRIME_FIRST_USED)
       static bool first_tool_is_primed = false;
-      if (new_tool == old_tool && !first_tool_is_primed && enable_first_prime) {
-        tool_change_prime();
+      
+      if (enable_first_prime && !first_tool_is_primed && new_tool == old_tool && new_tool == 0) {
+        if (!TEST(extruder_was_primed, old_tool)) {
+          tool_change_prime(); //Only perform this prime if it hasn't been primed yet!
+        }
```

Comment:
If you are looking for minor optimizations to save bits, the `first_tool_is_primed` bool can potentially be removed. 

Currently it's statically defined and only set once, and only to ensure this isn't processed again. 

The 'extruder_was_primed' test ( against extruder 0) on the following line can be used in place of it. Not sure how taxing a call like that is compared to checking a bool on its own.